tests/test_examples.py

```diff
@@ -0,0 +1,46 @@
+import glob
+import os
+import shutil
+
+import pytest
+
+from rapids_dependency_file_generator.rapids_dependency_file_generator import (
+    main as dfg,
+)
+
+CURRENT_DIR = os.path.dirname(__file__)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def clean_actual_files():
+    for root, _, _ in os.walk("tests"):
+        if os.path.basename(root) == "actual":
+            shutil.rmtree(root)
+
+
+def test1():
+    test_dir = os.path.join(CURRENT_DIR, "examples", "test1")
```

Comment:
You might like to try using `pathlib.Path` objects for most of the places you're using `os.path` in this repo. It's a slightly more modern and concise way of writing things -- but that's an optional refactoring.